python/lsst/meas/base/plugins.py

```diff
@@ -68,6 +71,103 @@
                     TransformClass=bl.ApertureFluxTransform, executionOrder=BasePlugin.FLUX_ORDER)
 
 # --- Single-Frame Measurement Plugins ---
+class SingleFrameVarianceConfig(SingleFramePluginConfig):
+        scale = lsst.pex.config.Field(dtype=float, default=5.0, optional=True,
+                                      doc="scale factor to apply to shape for aperture")
+        mask = lsst.pex.config.ListField(doc="mask planes to ignore", dtype=str,
+                                         default=["DETECTED", "DETECTED_NEGATIVE"])
+
+@register("base_Variance")
+class SingleFrameVariancePlugin(SingleFramePlugin):
+    '''
+    Calculate the median variance within a Footprint scaled from the object shape so
+    the value is not terribly influenced by the object and instead represents the
+    variance in the background near the object.
+    '''
+
+    ConfigClass = SingleFrameVarianceConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.varValue = schema.addField(name + '_value', type="D", doc="variance at object position")
+        self.varFlag = schema.addField(name + '_flag', type="Flag", doc="Set to 1 for any fatal Failure")
+
+    def measure(self, measRecord, exposure):
+        # Promote the bounding box of the footprint to type D to ensure
+        # the ability to compare (they may be of mixed types I and D)
+        center = measRecord.getCentroid()
+        fpBbox = lsst.afw.geom.Box2D(measRecord.getFootprint().getBBox())
+        # check to insure that the center exists and that it is contained within the footprint
+        # catch bad centroiding
+        if not center or not fpBbox.contains(center):
+            raise Exception("The measRecord has no center, or the center is outside the footprint")
+
+        # Create an aperture and grow it by scale value defined in config to ensure there are enough
+        # pixles around the object to get decent statistics
+        aperture = lsst.afw.geom.ellipses.Ellipse(measRecord.getShape(), measRecord.getCentroid())
+        aperture.scale(self.config.scale)
+        foot = lsst.afw.detection.Footprint(aperture)
+        foot.clipTo(exposure.getBBox(lsst.afw.image.PARENT))
+        # Filter out any pixels which have mask bits set corresponding to the planes to be excluded
+        # (defined in config.mask)
+        maskedImage = exposure.getMaskedImage()
+        maskBits = maskedImage.getMask().getPlaneBitMask(self.config.mask)
+        logicalMask = numpy.logical_not(maskedImage.getMask().getArray() & maskBits)
+        # Compute the median variance value for each pixel not excluded by the mask and write the record
+        medVar = numpy.median(maskedImage.getVariance().getArray()[logicalMask])
```

Comment:
Would it be better to use `lsst.afw.math.makeStatistics()` rather than NumPy? We've seen before that the latter is sometimes inconsistent between versions.